### PR TITLE
Dabonnie/add documentation

### PR DESCRIPTION
**Pull request by dabonnie**
Add description and documentation

### CODE REVIEW

File: system_metrics_collector/README.md [hunk 1]
```diff
@@ -19,7 +41,8 @@ Type: integer
 ```sh
```
Comment (zmichaels11) [inline system_metrics_collector/README.md]:
```suggestion
 ```yaml
```

File: system_metrics_collector/README.md [hunk 2]
```diff
@@ -29,20 +52,25 @@ Parameter name: publish_period
...
+### Manual Execution
+You can run the example main (todo rename as example) which will manually start the system CPU, system memory, process
+CPU and process memory measurement nodes.
 ```
 ros2 run system_metrics_collector main
 ```
 
 Change `publish_period` or `measurement_period` using `--ros-args`:
-
 ```
```
Comment (zmichaels11) [inline system_metrics_collector/README.md]:
```suggestion
 ```sh
```

File: system_metrics_collector/README.md [hunk 3]
```diff
@@ -51,15 +79,62 @@ ros2 node list
...
-Subscribe to `/system_metrics` to visualize all the statistics:
+## Interaction via the ROS2 CLI
+This section describes how to interact with this package using the
+[ROS2 Command Line Interface (CLI)](https://github.com/ros2/ros2cli). Example output is provided where relevant.
+
+
+#### Subscribe to `/system_metrics` to visualize all the statistics.
+Using [ros2topic](https://github.com/ros2/ros2cli/tree/master/ros2topic):
 
 ```
```
Comment (zmichaels11) [inline system_metrics_collector/README.md]:
```suggestion
 ```yaml
```

File: system_metrics_collector/README.md (hunk 2 above)
Comment (zmichaels11) [reply system_metrics_collector/README.md]: This suggestion applies to a few sections of the `README.md`

File: system_metrics_collector/README.md [hunk 4]
```diff
@@ -1,7 +1,29 @@
-# System metric collector
+# System metrics collector
 
-## Parameters
+## Supported targets
+The system metrics collector package supports the following:
+
+* ROS Distro
+  * ROS 2: `eloquent`
+* OS: `Ubuntu`
```
[the remark this answers is not in the document]
Comment (dabonnie) [reply system_metrics_collector/README.md]: Done

File: system_metrics_collector/README.md (hunk 1 above)
Comment (dabonnie) [reply system_metrics_collector/README.md]: Done

File: system_metrics_collector/README.md (hunk 2 above)
Comment (dabonnie) [reply system_metrics_collector/README.md]: Done here and applied to the rest of the README

File: README.md [hunk 5]
```diff
@@ -7,7 +7,18 @@ Note: _This is currently an **experimental** package._
...
 
-This message is used to publish the statistics of measured data points, e.g. system cpu %, system free memory, message age, etc.
+This message is used to publish the statistics of measured data points, for example system CPU percentage,
+system free memory percentage, message age, etc.
+
+## System Metrics Collector
+
+The goal of this package is to provide lightweight, real-time system metrics to enable
+system debugging and diagnosis of ROS2 systems (currently Linux only). It automatically collects
+and aggregates system CPU, system memory, as well as ROS2 process CPU and memory.
```
Comment (piraka9011) [inline README.md]: It automatically collects and aggregates CPU and memory usage of both system and ROS2 processes.

File: README.md [hunk 6]
```diff
@@ -7,7 +7,18 @@ Note: _This is currently an **experimental** package._
...
+Data is aggregated a in order to provide constant time average, min, max, sample count,
```
Comment (piraka9011) [inline README.md]: aggregated in

File: README.md [hunk 7]
```diff
@@ -7,7 +7,18 @@ Note: _This is currently an **experimental** package._
...
+and standard deviation values for each collected metric.
+
+Please see the [package README](https://github.com/ros-tooling/system_metrics_collector/blob/master/system_metrics_collector/README.md) for more details.
```
Comment (piraka9011) [inline README.md]: nit: use relative path

File: system_metrics_collector/README.md [hunk 8]
```diff
@@ -1,11 +1,33 @@
...
+
+* ROS Distro
+  * ROS 2: `eloquent`
+* OS: `Ubuntu Bionic`
+
+## Description
+This package aims to integrate lightweight aggregation tools in order to collect, measure, and publish aggregate metrics.
 
+The [moving average](https://github.com/ros-tooling/system_metrics_collector/blob/master/system_metrics_collector/src/moving_average_statistics/moving_average.hpp)
+tools provide constant time sample aggregation to produce average, min, max, standard deviation, and sample count. The
```
Comment (piraka9011) [inline system_metrics_collector/README.md]: Nit: new line after period and use relative paths for files.

File: system_metrics_collector/README.md [hunk 9]
```diff
@@ -29,37 +52,89 @@ Parameter name: publish_period
 
 ## Usage:
 
-You can run the following command to generate CPU and Memory statistics. for both, all the system and the node running:
-
+### ROS2 Launch
+The system CPU and system memory can be launched by using [ros2 launch](https://github.com/ros2/launch):
```
Comment (piraka9011) [inline system_metrics_collector/README.md]: ... and system memory nodes...

Comment (piraka9011) [reply system_metrics_collector/README.md]: can be launch with

File: README.md (hunk 6 above)
Comment (dabonnie) [reply README.md]: Done

File: system_metrics_collector/README.md (hunk 8 above)
Comment (dabonnie) [reply system_metrics_collector/README.md]: Done

File: system_metrics_collector/README.md (hunk 9 above)
Comment (dabonnie) [reply system_metrics_collector/README.md]: Done

**Comment by dabonnie** [comment]
Build successful, but codecov is still broken.

**Pull request merged**